website/addons/box/utils.py

```diff
@@ -78,6 +86,40 @@ def log(self, action, extra=None, save=False):
             self.node.save()
 
 
+def refresh_oauth_key(external_account, force=False):
```

Comment:
`refresh_v2_token` comes from the `box.py` client library. Though it does need to be generalized, I agree. I think that work should go in a separate PR, as `website.oauth` doesn't currently have any token-refreshing capabilities and this PR is for Box.